### DISCUSSION TITLE
Объединение массивов с созданием подмассива

### DISCUSSION THREAD

**Discussion by ToeraHL** [totumonline/totum-ru-issues-and-discussions#315] [Как это сделать?] [answered]
Здравствуйте. Подскажите, пожалуйста, как сгруппировать массивы так, чтобы получилось следующее:
**исходный**
{
  [
    {q : 77},
    {w : 33},
    {e : tt},
    {a : 1}
  ],
  [
    {q : 77},
    {w : 33},
    {e : tt},
    {a : 2}
  ],
  [
    {q : 77},
    {w : 33},
    {e : kk},
    {a : 3}
  ],
  [
    {q : 77},
    {w : 33},
    {e : kk},
    {a : 4}
  ],
}

**получилось**
{
  [
    {q : 77},
    {w : 33},
    {e : tt},
    {a : [1,2]}
  ],
  [
    {q : 77},
    {w : 33},
    {e : kk},
    {a : [3,4]}
  ]
}

Не могу сообразить, как объединить массивы по значению ключей "q", "w","e", собрать значения из "a" в подмассив и при этом не написать громоздкий тяжелый код

Хочется удалить "a", потом получить уникальные списки listUniq, и добавить "a" по условию, но непонятно как удалить элемент массива по ключу

Заранее большое спасибо

**Comment by ttm-alexeypolunin** [comment]
По удалить смотрите rowKeysRemove — https://ru.docs.totum.online/functions#fn-rowKeysRemove

По остальному я посмотрю завтра

**Reply by ToeraHL** [reply]
Спасибо. Попробую. Если получится решить задачку, отпишусь

**Comment by ToeraHL** [comment, marked as answer]
Сделала, но не так как предполагала - в разных ячейках

1. Сначала удалила элемент, который не нужно сравнивать и выбрала уникальные
`listUniq (list: $key_minus)`
`        key_minus: rowKeysRemove(row: #h_data_insert; key: "a"; recursive: true)`

2. Загрузила получение в строки колонки data
`insertList (table: $#ntn; field: 'data' = $#nfv; log: true)`

В каждой из строк получилось примерно такое: 
первая строка
{q : 77},
{w : 33},
{e : tt},
вторая строка
{q : 77},
{w : 33},
{e : kk},

Создала рядом еще одну колонку и там сравнила полученое с изначальным массивом:
`listReplace(list: #h_data_insert; action: $comparison; value: "value")`
`        comparison: if (condition: $#value["q"] == #data["q"]; condition: $#value["w"] == #data["w"]; condition: $#value["e"] == #data["e"];  then: $a; else: )`
`                a: $#value["a"]`
		
Мне удалось выделить в массив объединенные элементы. Мою задачу решает. Думаю, все эти этапы, которые я делала в разных ячейках при надобности можно собрать в одну
Не знаю на сколько это хорошее решение по нагрузке. Данных мало, чтобы было ощутимо

**Reply by ttm-alexeypolunin** [reply]
Я думаю, что оптимал схема такая.

Добавляем прям в оперативке через listReplace еще одну колонку, куда клеим из q-w-e значение.

По этой колонке берем уникальные — получаем список основу.

Его суем в listReplace и восстанавливаем q/w/e и формируем новую a.
